# 디지털 사회에서 프라이버시는 계층적으로 거래되는가?

### note

메모: 가설1 - 객관적 위치 / **가설2 - 주관적 계층 위치**   

            방향으로..??? (종속변수가 willingness니까)

단순히 “개인정보 보호 인식이 높은가 낮은가”가 아니라, 디지털 사회에서 개인정보 제공이 계층적 조건과 연결되는가를 묻는 연구.

- 개인정보 제공은 개인의 ‘선택’처럼 보이지만, 실제로는 경제적 압박 속에서 이루어지는 선택일 수 있다.  
- 저소득층은 할인, 포인트, 무료상품, 서비스 접근성 때문에 더 쉽게 개인정보를 제공할 수 있다.  
- 따라서 개인정보 제공은 단순한 기술 수용 문제가 아니라 디지털 불평등, 데이터 착취, 프라이버시의 계층화 문제일 수 있다.  


즉, *디지털 자본주의에서 프라이버시가 어떻게 계층적으로 거래되는가?*
> 경제적 자원이 부족한 사람들은 개인정보를 보호하고 싶지 않아서가 아니라, 디지털 시장에서 더 작은 보상에도 개인정보 제공을 고려해야 하는 조건에 놓여 있을 수 있다.


문제점: 이론적 개념화가 채워져야 한다.  
> 프라이버시 불평등/데이터 착취 프레임(이론) 즉, 경제적 조건이 개인정보 제공의 선택지를 불평등하게 만든다는 방향으로 가는게 좋을듯.




## 종속변수에 대해

현재 종속변수는 2개이다. 

**GIVEPIFR: 할인이나 무료상품을 받기 위해 개인정보를 제공할 것** -> **현재 편향, 단기적인 이익, 데이터 경제 (할인이나 무료상품 같은 현재 편향!!)**/ 또한, 현재 편향에 대한 이론을 발전시킬 수 있을 것이다.   
GIVEPIPC: 민간기업이 개인정보를 사용하여 이익을 내더라도 제공할 것 (-> 나중에 어떤 추가적인 분석에서 사용할 수는 있을거 같음)

> 우리의 연구 질문과 잘 맞는 것은 GIVEPIFR이다. 왜냐하면 GIVEPIPC는 기업이 내 개인정보로 이익을 얻는 것에 대한 허용성에 가깝습니다. 즉, 개인정보 제공 의향이라기보다는 기업의 데이터 활용에 대한 태도, 혹은 데이터 상업화 수용성에 가깝다. 
    **단, GIVEPIFR를 종속변수로 설정한 이유를 밝히는데 주목하자** 

따라서 현재 연구질문에는 종속변수를 GIVEPIFR로 가는 것이 더 타당해 보인다. 가능하다면, 보조연구로 가능할 지도...? (나에게 오는 혜택을 위한 제공과 기업 이익을 위한 제공이 계층에 따라 다르게 나타나는지 비교해보는 방향으로)


### 독립변수에 대해

우리 연구의 핵심 독립변수는 다음과 같다. 

log_INCOM0: 주 분석의 객관적 경제 자원 -> 객관적   (범주형 자료를 쓸 지..?)  -> **여쭤보자....**    

RANK: 주관적 계층의식 -> 주관적  
        이성적으로 판단하는 경향이 있을 것임. 

SATFIN_R: 가계상태 만족도 역코딩 변수 -> 주관적  
        하지만, 만족도는 더 주관성이 강하다. 

FINPROS_R: 가계경제상태 전망 역코딩 변수 -> 주관적  
        하지만, 만족도는 더 주관성이 강하다. 

-> 설명을 조금씩 다르게 할 수 있다. 큰 범위에서 개인의 주관적 평가이지만, 다른 설명이 가능할 것임. 


그런데 이 변수들은 경제상태라는 점에서 매우 강하게 연관이 있어 보인다. 이 변수를 한번에 넣고 통제하면 소득의 전체 효과를 스스로 지워버리는게 아닐까...? (과잉통제)

그래서 아래와 같이 모형을 위계적으로 나눌 수도 있을거 같다. 

| 모형      | 목적                           |
| ------- | ---------------------------- |
| Model 1 | 객관적 계층: 소득만 투입               |
| Model 2 | 인구사회학적 통제 추가: 연령, 성별, 교육     |
| Model 3 | 주관적 계층 변수 추가: RANK           |
| Model 4 | 경제 심리 변수 추가: SATFIN_R |
| Model 5 | FINPROS_R |
| Model 6 | *디지털 역량/기술위험 인식 추가*            |  

또는 Study1: 주관적 계층의식에 대한 변수만 (RANK, SATFIN_R, FINPROS_R)  
Study2: 객관적 계층의식은 주 분석에서 log_INCOM0로 검정하고, INCOME 범주형 변수는 강건성 분석에서 비교


추가로 INCOM0: 월평균 가구소득(원료/연속형)
INCOME: 월평균가구소득(범주형)   -> 이거 두 개가 다른 버전으로 분석을 했는데, 주 분석 변수를 INCOM0으로 하고, INCOME 범주형은 강건성 검토(소득 측정 방식이 달라져도 결과가 유지되는지 확인하는 용도)로 쓰이는 것을 고려해야함. 

> 연속형을 쓰는게 나은 이유: 정보 손실이 적다. 범주형은 묶은 변수니까 정보손실 우려가 있음. 또한, 연속형 변수가 회귀모형 해석에 더 간결함. 범주형 쓰면 독변 여러개가 되니까. (물론, INCOM0을 그대로 쓰지 않고 로그변환을 해서 쓴다. 소득 변수는 보통 분포가 한쪽으로 치우쳐져 있어서 로그 변환!)

*통제변수에 대해*

AGE  
SEX  
EDUC   
ABLITINF_R  
TECHHARM_R  

여기서 ABLITINF_R와 TECHHARM_R를 통제변수로 넣는게 적당할까 아니면 *조절변수* 또는 독립변수로 하는게 적당할까?  
> ABLITINF_R와 TECHHARM_R는 오히려 계층의 결과일 수 있으니까.. 이 경우 ABLITINF_R는 매개변수일 수 있고

이런 고려도 가능할듯

| 모형      | 투입 변수                  | 목적                             |
| ------- | ---------------------- | ------------------------------ |
| Model 1 | 계층 변수만                 | 계층과 개인정보 제공 태도의 기본 관계 확인       |
| Model 2 | 계층 + 인구사회학적 통제         | 연령, 성별, 교육을 고려해도 계층 효과가 남는지 확인 |
| Model 3 | Model 2 + `ABLITINF_R`   | 디지털 정보판별능력을 고려해도 계층 효과가 남는지 확인 |   
| Model 4 | Model 3 + `TECHHARM_R` | 기술위험 인식까지 고려해도 계층 효과가 남는지 확인   |
-> 연구 살펴보면서 추가로 고민해야할 부분이 아닌가...

만약,


경우 A: 계층 효과가 계속 남는다

ABLITINF_R, TECHHARM_R를 통제한 뒤에도 계층 변수의 효과가 유지된다.

이 경우 해석은 강해짐. 

개인정보 제공 태도의 계층 차이는 단순히 디지털 리터러시나 기술위험 인식 차이로만 설명되지 않는다. 계층적 위치 자체가 개인정보 제공 태도와 독립적으로 관련되어 있다.


경우 B: 계층 효과가 약해지지만 남는다

ABLITINF_R, TECHHARM_R를 넣으면 계층 효과가 줄어들지만 여전히 유의하다.

계층과 개인정보 제공 태도의 관계 일부는 역코딩된 디지털 정보판별능력이나 기술위험 인식 차이를 통해 설명되지만, 그것만으로 완전히 설명되지는 않는다.


경우 C: 계층 효과가 사라진다

ABLITINF_R, TECHHARM_R를 넣으면 계층 효과가 사라진다.

개인정보 제공 태도의 계층 차이는 계층 그 자체보다, 계층에 따라 다르게 형성된 디지털 정보판별능력과 기술위험 인식에 의해 설명될 가능성이 있다.


> 본 연구는 계층적 위치가 개인정보 제공 동의에 미치는 영향을 분석하되, 이 관계가 단순히 디지털 정보판별능력이나 기술에 대한 위험 인식의 차이로 환원되는지를 검토하였다. 이를 위해 기본 모형에서는 객관적·주관적 계층 변수와 인구사회학적 통제변수를 투입하고, 확장 모형에서 ABLITINF_R와 TECHHARM_R를 단계적으로 추가하였다. 이 접근은 계층 효과의 독립성과 설명 경로를 동시에 확인하기 위한 것이다.


# 디지털 사회에서 개인정보 제공 태도는 계층에 따라 달라지는가?

**연구질문**: 경제적 자원이 낮은 사람일수록 할인이나 혜택을 위해 개인정보를 제공할 의향이 높은가?

이 노트북은 KOSSDA `adt_2025_drop_0509.csv` 자료를 사용해 `statsmodels` 기반 회귀분석을 수행한다.

중요한 해석상 주의:
- 이 자료는 관찰자료이므로 회귀계수는 조건부 상관관계로 해석하는 것이 안전하다. 엄밀한 인과효과 주장을 위해서는 무작위 배정, 자연실험, 패널/도구변수 등 추가 식별전략이 필요하다.


## 가설

H1. 객관적 경제 자원이 낮을수록 할인·무료상품을 위한 개인정보 제공 의향은 높을 것이다.

H2. 주관적 계층의식이 낮을수록 할인·무료상품을 위한 개인정보 제공 의향은 높을 것이다.

H3. 가계경제상태에 대한 만족도와 전망이 낮을수록 할인·무료상품을 위한 개인정보 제공 의향은 높을 것이다.

H4. 디지털 정보판별능력(`ABLITINF_R`)과 기술위험 인식(`TECHHARM_R`)은 경제적 자원과 개인정보 제공 의향의 관계를 약화 또는 강화할 수 있다.(물론, 이 가설은 ABLITINF_R와 TECHHARM_R를 조절변수로 보았을 때이다. 즉, 경제적 자원이 개인정보 제공 의향에 미치는 효과가 ABLITINF_R와 TECHHARM_R 수준에 따라 달라진다.)

## 회귀식

주 분석에서는 객관적 계층 변수를 `INCOM0`의 로그변환값인 `log_INCOM0`로 사용한다. `INCOM0`는 월평균 가구소득 원자료/연속형 변수이므로, 소득 분포의 왜도와 극단값 영향을 줄이기 위해 `log1p(INCOM0)`로 변환해 투입한다.

종속변수는 KGSS 원척도에서 `1=매우 동의`, `5=매우 반대`로 측정되어 있다. 본 연구는 개인정보 제공에 대한 **동의 수준**을 해석하고자 하므로, 분석에서는 이를 역코딩한 `GIVEPIFR_R`, `GIVEPIPC_R`을 사용한다.

- 원척도: 1=매우 동의, 2=동의, 3=동의도 반대도 아님, 4=반대, 5=매우 반대
- 분석척도: 1=매우 반대, 2=반대, 3=동의도 반대도 아님, 4=동의, 5=매우 동의

`SATFIN`, `FINPROS`, `ABLITINF`도 해석 방향을 맞추기 위해 역코딩한 분석 변수 `SATFIN_R`, `FINPROS_R`, `ABLITINF_R`을 사용한다. 값이 높을수록 각각 가계상태 만족도, 가계경제 전망의 긍정성, 정보판별능력이 높다. `TECHHARM_R`은 값이 높을수록 기술위험 인식이 높다는 뜻이다.

주요 선형 회귀식은 다음과 같다.

$$
Y_i = \beta_0 + \beta_1 \log(\mathrm{INCOM0}_i + 1) + \beta_2 \mathrm{RANK}_i + \beta_3 \mathrm{SATFIN\_R}_i + \beta_4 \mathrm{FINPROS\_R}_i + \gamma X_i + \epsilon_i
$$

여기서 종속변수 $Y_i$는 각각 다음 두 문항을 역코딩한 변수다.

- `GIVEPIFR_R`: 할인이나 무료상품을 받기 위해 개인정보를 제공할 의향. 값이 높을수록 동의가 강함
- `GIVEPIPC_R`: 민간기업이 개인정보를 사용하여 이익을 내더라도 제공할 의향. 값이 높을수록 동의가 강함

핵심 독립변수는 `log_INCOM0`, `RANK`, `SATFIN_R`, `FINPROS_R`이며, 통제변수 $X_i$에는 `AGE`, `SEX`, `EDUC`, `ABLITINF_R`, `TECHHARM_R`이 포함된다.  
`INCOME`은 월평균 가구소득 범주형 변수이므로 주 분석에는 넣지 않고, 연구 마지막의 강건성 분석에서 소득 측정 방식을 바꿔도 결과가 유지되는지 확인하는 용도로 사용한다.

추가로 5점 척도를 이진화한 로짓모형도 추정한다.

$$
\Pr(\mathrm{Agree}_i = 1) = \operatorname{logit}^{-1}\left(\beta_0 + \beta_1 \log(\mathrm{INCOM0}_i + 1) + \beta_2 \mathrm{RANK}_i + \beta_3 \mathrm{SATFIN\_R}_i + \beta_4 \mathrm{FINPROS\_R}_i + \gamma X_i\right)
$$

회귀분석과 로짓 분석에 대해 이야기해 보자면

> 1. 선형 회귀분석
>> 원척도를 역코딩한 1~5점 점수를 사용한다. 값이 높을수록 개인정보 제공에 더 동의한다.

> 2. 로짓분석
>> 원척도 기준 1 또는 2, 즉 "매우 동의" 또는 "동의"를 1로 코딩하고, 3~5는 0으로 코딩한다.

  | 원래 응답값 | 원래 의미 | 역코딩 후 | 로짓 이진화 |
  |---:|---|---:|---:|
  | 1 | 매우 동의 | 5 | 1 |
  | 2 | 동의 | 4 | 1 |
  | 3 | 동의도 반대도 아님 | 3 | 0 |
  | 4 | 반대 | 2 | 0 |
  | 5 | 매우 반대 | 1 | 0 |

이진화한 대상 변수는 다음 두 개입니다.

| 원래 변수 | 역코딩 변수 | 이진화 변수 | 의미 |
|---|---|---|---|
| GIVEPIFR | GIVEPIFR_R | GIVEPIFR_agree | 할인/무료상품을 받기 위해 개인정보를 제공할 의향 있음 |
| GIVEPIPC | GIVEPIPC_R | GIVEPIPC_agree | 민간기업이 개인정보로 이익을 내더라도 제공할 의향 있음 |

결론  
  - 선형 회귀분석: 역코딩된 1~5점 점수가 높아지는지 분석한다. 점수가 높을수록 개인정보 제공 동의가 높다.
  - 로짓분석: 1~2점, 즉 “동의/매우 동의”일 확률이 높아지는지 분석한다.

In [16]:
import warnings
warnings.filterwarnings("ignore")

from pathlib import Path
import numpy as np
import pandas as pd
import statsmodels.api as sm
import statsmodels.formula.api as smf
from IPython.display import display

In [17]:
DATA_PATH = Path(r"/home/pck/data_jour/sda_Team Project_/adt_2025_drop_0509.csv")
df_raw = pd.read_csv(DATA_PATH)

print(df_raw.shape)
df_raw.head()

(1211, 455)


,YEAR,RESPID,YRRESPID,FINALWT,SEX,AGE,MARITAL,EMPLY,WHYNOE,EDUC,...,HHWHYNOE,HHDNO,HOMPOP,SEPAPOP,UNRELAT,REGION,URBAN,SAMPLEAB,INTDATM,INTDATD
0,2025.0,101.0,202500101.0,1.349140,1.0,44.0,1.0,1.0,-1.0,4.0,...,-1.0,11.0,3.0,0.0,0.0,1.0,1.0,1.0,4.0,6.0
1,2025.0,102.0,202500102.0,0.719461,2.0,62.0,1.0,2.0,3.0,3.0,...,-1.0,2.0,2.0,0.0,0.0,1.0,1.0,2.0,4.0,6.0
2,2025.0,103.0,202500103.0,0.572766,2.0,50.0,1.0,2.0,18.0,3.0,...,-1.0,4.0,3.0,0.0,0.0,1.0,1.0,2.0,4.0,7.0
3,2025.0,104.0,202500104.0,1.093403,1.0,62.0,1.0,1.0,-1.0,3.0,...,-1.0,10.0,2.0,0.0,0.0,1.0,1.0,2.0,5.0,25.0
4,2025.0,105.0,202500105.0,1.093403,1.0,74.0,1.0,2.0,19.0,2.0,...,19.0,12.0,2.0,0.0,0.0,1.0,1.0,2.0,4.0,8.0


In [18]:
# 분석에 사용할 변수 전처리
raw_dv_vars = ["GIVEPIFR", "GIVEPIPC"]
dv_vars = ["GIVEPIFR_R", "GIVEPIPC_R"]
raw_core_vars = ["INCOME", "INCOM0", "RANK", "SATFIN", "FINPROS"]
core_vars = ["INCOME", "INCOM0", "RANK", "SATFIN_R", "FINPROS_R"]
raw_control_vars = ["AGE", "SEX", "EDUC", "ABLITINF", "TECHHARM"]
control_vars = ["AGE", "SEX", "EDUC", "ABLITINF_R", "TECHHARM_R"]
weight_var = "FINALWT"
analysis_vars_raw = raw_dv_vars + raw_core_vars + raw_control_vars + [weight_var]

missing = [v for v in analysis_vars_raw if v not in df_raw.columns]
if missing:
    raise ValueError(f"데이터에 없는 변수: {missing}")

df_raw[analysis_vars_raw].describe().T

,count,mean,std,min,25%,50%,75%,max
GIVEPIFR,1211.0,3.507019,1.121057,-8.000000,3.000000,4.00000,4.000000,5.000000
GIVEPIPC,1211.0,3.824112,1.022691,-8.000000,3.000000,4.00000,5.000000,5.000000
INCOME,1211.0,8.895128,6.527101,-8.000000,5.000000,9.00000,13.000000,21.000000
INCOM0,1211.0,448.386457,407.973516,-8.000000,216.000000,400.00000,600.000000,9000.000000
RANK,1211.0,5.065235,1.498717,1.000000,4.000000,5.00000,6.000000,10.000000
SATFIN,1211.0,3.182494,0.886384,1.000000,3.000000,3.00000,4.000000,5.000000
FINPROS,1211.0,2.777870,0.988047,-8.000000,2.000000,3.00000,3.000000,5.000000
AGE,1211.0,52.611891,15.651251,18.000000,40.000000,54.00000,64.000000,93.000000
SEX,1211.0,1.610239,0.487897,1.000000,1.000000,2.00000,2.000000,2.000000
EDUC,1211.0,3.469860,1.483142,-8.000000,3.000000,3.00000,5.000000,8.000000


*참고하세요!*

  | 코드상 구분 | 변수명 | 의미 |
  |---|---|---|
  | 원 종속변수 | GIVEPIFR | 개인정보 제공에 대한 태도1: 할인이나 무료상품을 받기 위해 제공할 것. 원척도는 1=매우 동의, 5=매우 반대 |
  | 원 종속변수 | GIVEPIPC | 개인정보 제공에 대한 태도2: 민간기업이 나의 개인정보를 사용하여 이익을 내더라도 제공할 것. 원척도는 1=매우 동의, 5=매우 반대 |
  | 분석 종속변수 | GIVEPIFR_R | GIVEPIFR 역코딩 변수: 1=매우 반대, 5=매우 동의 |
  | 분석 종속변수 | GIVEPIPC_R | GIVEPIPC 역코딩 변수: 1=매우 반대, 5=매우 동의 |
  | 로짓 종속변수 | GIVEPIFR_agree | GIVEPIFR 원척도 1 또는 2를 1로 코딩한 제공 동의 여부 |
  | 로짓 종속변수 | GIVEPIPC_agree | GIVEPIPC 원척도 1 또는 2를 1로 코딩한 제공 동의 여부 |
  | 독립변수 | INCOME | 월평균 가구소득, 범주형 |
  | 독립변수 | INCOM0 | 월평균 가구소득, 원자료/연속형, 단위=만원 |
  | 독립변수 | RANK | 주관적인 계층의식, 10점 척도. 1=최하층, 10=최상층 |
  | 원 독립변수 | SATFIN | 가계상태 만족도 원척도. 1=매우 만족, 5=매우 불만족 |
  | 분석 독립변수 | SATFIN_R | SATFIN 역코딩 변수. 1=매우 불만족, 5=매우 만족 |
  | 원 독립변수 | FINPROS | 가계경제상태 전망 원척도. 1=상당히 좋아질 것이다, 5=상당히 나빠질 것이다 |
  | 분석 독립변수 | FINPROS_R | FINPROS 역코딩 변수. 1=상당히 나빠질 것이다, 5=상당히 좋아질 것이다 |
  | 통제변수 | AGE | 응답자 연령, 실제 만 나이 |
  | 통제변수 | SEX | 응답자 성별. 1=남자, 2=여자 |
  | 통제변수 | EDUC | 응답자 학력: 0=무학, 1=초등학교, 2=중학교, 3=고등학교, 4=전문대학, 5=대학교, 6=대학원 석사, 7=대학원 박사, 8=서당한학 |
  | 원 통제변수 | ABLITINF | 인터넷 활용능력3: 인터넷에서 공유하면 안 되는 정보를 구별할 수 있는 것. 1=매우 높음, 5=매우 낮음 |
  | 분석 통제변수 | ABLITINF_R | ABLITINF 역코딩 변수. 1=매우 낮음, 5=매우 높음 |
  | 통제변수 | TECHHARM_R | 기술위험 인식 역코딩 변수: 값이 높을수록 기술이 이익보다 해를 더 많이 끼친다고 인식. 역코딩 전 TECHHARM은 1=매우 동의, 5=매우 반대 |
  | 가중치 | FINALWT | 최종 가중치 |

FINALWT 변수에 대해서 설명해 드리자면, 표본가중치 변수입니다. 즉, 설문에 응답한 한 사람이 실제 모집단에서 몇 명을 대표하는지를 반영하는 값입니다.

  예를 들어 설문조사 표본에 어떤 집단이 실제 인구보다 적게 뽑혔다면, 그 집단 응답자에게 더 큰 가중치를 줍니다. 반대로 어떤 집단이 실제
  인구보다 많이 뽑혔다면, 그 집단 응답자에게 더 작은 가중치를 줍니다.


 회귀분석에서는 FINALWT를 사용해서 이렇게 분석했습니다.

  smf.wls(formula=formula, data=model_data, weights=model_data["FINALWT"])

  이건 일반 OLS가 아니라 가중회귀분석 WLS입니다.

  의미는:

  > 가중치가 큰 응답자는 분석에서 더 큰 비중을 갖고, 가중치가 작은 응답자는 더 작은 비중을 갖는다.  
  > 본 분석에서는 표본의 대표성을 보정하기 위해 조사자료에서 제공하는 최종가중치 변수를 적용한 가중회귀분석을 실시한다! (로 쓰면 될 거 같습니다)

## 분석용 데이터 생성

처리 원칙:
- `-8`은 결측으로 처리한다.
- `ABLITINF` 원변수의 `-1`은 비해당 값으로 보이므로 결측으로 처리한다.
- `GIVEPIFR`와 `GIVEPIPC`의 원척도는 `1=매우 동의`, `5=매우 반대`이다.
- 선형회귀/WLS에서는 해석을 쉽게 하기 위해 역코딩 변수 `GIVEPIFR_R`, `GIVEPIPC_R`을 사용한다. 값이 높을수록 개인정보 제공 동의가 높다.
- `SATFIN`, `FINPROS`, `ABLITINF`도 해석 방향을 맞추기 위해 각각 `SATFIN_R`, `FINPROS_R`, `ABLITINF_R`로 역코딩한다. 값이 높을수록 각각 가계상태 만족, 가계경제 전망 긍정, 정보판별능력이 높다.
- 로짓모형용 `GIVEPI*_agree`는 원척도 기준 `1` 또는 `2`를 동의/의향 있음으로 보고 1로 코딩한다. `3`, `4`, `5`는 0으로 코딩한다.

In [19]:
df = df_raw[analysis_vars_raw].copy()

# 공통 결측 코드 처리
for col in analysis_vars_raw:
    df[col] = df[col].replace(-8, np.nan)

# ABLITINF의 -1은 비해당으로 보이므로 주 모형에서는 결측 처리
df["ABLITINF"] = df["ABLITINF"].replace(-1, np.nan)

# TECHHARM 역코딩: 원점수는 낮을수록 기술위험 인식이 높으므로, 분석에서는 높을수록 위험 인식이 높게 변환
# 원점수 1=매우 동의, 2=동의, 3=중립, 4=반대, 5=매우 반대
# 역코딩 5=매우 동의(위험 인식 높음), 1=매우 반대(위험 인식 낮음)
df["TECHHARM_R"] = 6 - df["TECHHARM"]

# 개인정보 제공 태도 역코딩: 원척도는 1=매우 동의, 5=매우 반대
# 분석척도는 1=매우 반대, 5=매우 동의가 되도록 변환
for y in raw_dv_vars:
    df[f"{y}_R"] = np.where(df[y].isin([1, 2, 3, 4, 5]), 6 - df[y], np.nan)

# 핵심 독립변수/통제변수 역코딩: 값이 높을수록 개념이 긍정적/높은 방향이 되도록 변환
# SATFIN_R: 1=매우 불만족, 5=매우 만족
# FINPROS_R: 1=상당히 나빠질 것, 5=상당히 좋아질 것
# ABLITINF_R: 1=매우 낮음, 5=매우 높음
for v in ["SATFIN", "FINPROS", "ABLITINF"]:
    df[f"{v}_R"] = np.where(df[v].isin([1, 2, 3, 4, 5]), 6 - df[v], np.nan)

analysis_vars = dv_vars + core_vars + control_vars + [weight_var]

# 로짓모형용 이진 종속변수: 원척도 1=매우 동의, 2=동의를 1로 코딩
# 3=동의도 반대도 아님, 4=반대, 5=매우 반대는 0으로 코딩
for y in raw_dv_vars:
    df[f"{y}_agree"] = np.where(df[y].isin([1, 2]), 1,
                         np.where(df[y].isin([3, 4, 5]), 0, np.nan))

# 연속형 소득은 극단값 영향을 줄이기 위해 로그 변환도 생성
df["log_INCOM0"] = np.log1p(df["INCOM0"].where(df["INCOM0"] >= 0, np.nan))

# INCOME은 범주형 소득구간 변수다. 원범주가 22개이고 일부 구간 표본이 작아,
# 회귀/로짓 모형에서는 인접 구간을 묶은 범주형 변수로 사용한다.
income_cat_bins = [-0.1, 2, 5, 8, 11, 14, 17, 21]
income_cat_labels = ["0-2", "3-5", "6-8", "9-11", "12-14", "15-17", "18-21"]
df["INCOME_CAT"] = pd.cut(
    df["INCOME"],
    bins=income_cat_bins,
    labels=income_cat_labels,
    include_lowest=True
)

# 표준화 변수: 계수 크기 비교용
## 서로 단위 다른 변수들 비교 가능하도록 같은 척도로 바꾼 변수입니다. 
## 단위가 서로 다른 독립변수들을 평균 0, 표준편차 1로 맞춘 뒤, 어떤 변수가 개인정보 제공 태도와 더 강하게 관련되는지 비교하기 위해서 표준화했습니다~
# INCOME은 범주형 소득구간 변수이므로 표준화 대상에서 제외하고, 연속형/순서형 분석 변수만 표준화
scale_vars = ["INCOM0", "log_INCOM0", "RANK", "SATFIN_R", "FINPROS_R", "AGE", "ABLITINF_R", "TECHHARM_R"]
for v in scale_vars:
    m = df[v].mean(skipna=True)
    s = df[v].std(skipna=True)
    df[f"z_{v}"] = (df[v] - m) / s

print(df[analysis_vars + raw_dv_vars + raw_core_vars + raw_control_vars + ["GIVEPIFR_agree", "GIVEPIPC_agree", "log_INCOM0", "INCOME_CAT"]].isna().sum())
display(df[["GIVEPIFR", "GIVEPIFR_R", "GIVEPIFR_agree", "GIVEPIPC", "GIVEPIPC_R", "GIVEPIPC_agree"]].head())
display(df[["SATFIN", "SATFIN_R", "FINPROS", "FINPROS_R", "ABLITINF", "ABLITINF_R", "TECHHARM", "TECHHARM_R"]].head())
display(df.head())
display(df.columns)

GIVEPIFR_R         1
GIVEPIPC_R         1
INCOME            60
INCOM0            60
RANK               0
SATFIN_R           0
FINPROS_R          4
AGE                0
SEX                0
EDUC               1
ABLITINF_R        88
TECHHARM_R         1
FINALWT            0
GIVEPIFR           1
GIVEPIPC           1
INCOME            60
INCOM0            60
RANK               0
SATFIN             0
FINPROS            4
AGE                0
SEX                0
EDUC               1
ABLITINF          88
TECHHARM           1
GIVEPIFR_agree     1
GIVEPIPC_agree     1
log_INCOM0        60
INCOME_CAT        60
dtype: int64


,GIVEPIFR,GIVEPIFR_R,GIVEPIFR_agree,GIVEPIPC,GIVEPIPC_R,GIVEPIPC_agree
0,3.0,3.0,0.0,2.0,4.0,1.0
1,2.0,4.0,1.0,3.0,3.0,0.0
2,3.0,3.0,0.0,3.0,3.0,0.0
3,2.0,4.0,1.0,2.0,4.0,1.0
4,4.0,2.0,0.0,3.0,3.0,0.0


,SATFIN,SATFIN_R,FINPROS,FINPROS_R,ABLITINF,ABLITINF_R,TECHHARM,TECHHARM_R
0,4.0,2.0,4.0,2.0,2.0,4.0,2.0,4.0
1,4.0,2.0,4.0,2.0,2.0,4.0,4.0,2.0
2,3.0,3.0,2.0,4.0,2.0,4.0,2.0,4.0
3,4.0,2.0,4.0,2.0,2.0,4.0,3.0,3.0
4,5.0,1.0,4.0,2.0,3.0,3.0,3.0,3.0


,GIVEPIFR,GIVEPIPC,INCOME,INCOM0,RANK,SATFIN,FINPROS,AGE,SEX,EDUC,...,log_INCOM0,INCOME_CAT,z_INCOM0,z_log_INCOM0,z_RANK,z_SATFIN_R,z_FINPROS_R,z_AGE,z_ABLITINF_R,z_TECHHARM_R
0,3.0,2.0,13.0,600.0,4.0,4.0,4.0,44.0,1.0,4.0,...,6.398595,12-14,0.315932,0.631253,-0.710765,-0.922293,-1.540774,-0.550237,0.562723,1.425217
1,2.0,3.0,6.0,280.0,4.0,4.0,4.0,62.0,2.0,3.0,...,5.638355,6-8,-0.474993,-0.317420,-0.710765,-0.922293,-1.540774,0.599831,0.562723,-0.717035
2,3.0,3.0,6.0,250.0,5.0,3.0,2.0,50.0,2.0,3.0,...,5.525453,6-8,-0.549142,-0.458306,-0.043527,0.205886,1.056592,-0.166881,0.562723,1.425217
3,2.0,2.0,NaN,NaN,2.0,4.0,4.0,62.0,1.0,3.0,...,NaN,NaN,NaN,NaN,-2.045239,-0.922293,-1.540774,0.599831,0.562723,0.354091
4,4.0,3.0,7.0,320.0,3.0,5.0,4.0,74.0,1.0,2.0,...,5.771441,6-8,-0.376128,-0.151347,-1.378002,-2.050472,-1.540774,1.366543,-0.323585,0.354091


Index(['GIVEPIFR', 'GIVEPIPC', 'INCOME', 'INCOM0', 'RANK', 'SATFIN', 'FINPROS',
       'AGE', 'SEX', 'EDUC', 'ABLITINF', 'TECHHARM', 'FINALWT', 'TECHHARM_R',
       'GIVEPIFR_R', 'GIVEPIPC_R', 'SATFIN_R', 'FINPROS_R', 'ABLITINF_R',
       'GIVEPIFR_agree', 'GIVEPIPC_agree', 'log_INCOM0', 'INCOME_CAT',
       'z_INCOM0', 'z_log_INCOM0', 'z_RANK', 'z_SATFIN_R', 'z_FINPROS_R',
       'z_AGE', 'z_ABLITINF_R', 'z_TECHHARM_R'],
      dtype='str')

In [20]:
# 기술통계
summary_vars = [
    "GIVEPIFR", "GIVEPIFR_R", "GIVEPIFR_agree",
    "GIVEPIPC", "GIVEPIPC_R", "GIVEPIPC_agree",
    "INCOME", "INCOM0", "log_INCOM0", "RANK",
    "SATFIN", "SATFIN_R", "FINPROS", "FINPROS_R",
    "AGE", "SEX", "EDUC", "ABLITINF", "ABLITINF_R", "TECHHARM", "TECHHARM_R"
]
df[summary_vars].describe().T

,count,mean,std,min,25%,50%,75%,max
GIVEPIFR,1210.0,3.516529,1.071539,1.0,3.000000,4.000000,4.000000,5.000000
GIVEPIFR_R,1210.0,2.483471,1.071539,1.0,2.000000,2.000000,3.000000,5.000000
GIVEPIFR_agree,1210.0,0.212397,0.409173,0.0,0.000000,0.000000,0.000000,1.000000
GIVEPIPC,1210.0,3.833884,0.964897,1.0,3.000000,4.000000,5.000000,5.000000
GIVEPIPC_R,1210.0,2.166116,0.964897,1.0,1.000000,2.000000,3.000000,5.000000
GIVEPIPC_agree,1210.0,0.100000,0.300124,0.0,0.000000,0.000000,0.000000,1.000000
INCOME,1151.0,9.775847,5.399692,0.0,6.000000,9.000000,13.000000,21.000000
INCOM0,1151.0,472.177237,404.589441,0.0,250.000000,400.000000,630.000000,9000.000000
log_INCOM0,1151.0,5.892726,0.801372,0.0,5.525453,5.993961,6.447306,9.105091
RANK,1211.0,5.065235,1.498717,1.0,4.000000,5.000000,6.000000,10.000000


In [ ]:
# 상관관계: 핵심 경제자원 변수와 개인정보 제공 동의 태도 (회귀 분석 전 상관관계 확인)
import matplotlib.pyplot as plt
import seaborn as sns

# INCOM0는 연속형 변수이므로 Pearson 상관계수에 포함
# INCOME은 범주형 소득구간 변수이므로 Pearson 상관계수에서 제외하고, 아래에서 범주별 평균으로 확인
# GIVEPI*_R은 값이 높을수록 개인정보 제공 동의가 높도록 역코딩한 분석용 종속변수
corr_vars = ["GIVEPIFR_R", "GIVEPIPC_R", "INCOM0", "RANK", "SATFIN_R", "FINPROS_R"]
corr_matrix = df[corr_vars].corr(method="pearson")

display(corr_matrix)

plt.figure(figsize=(9, 7))
sns.heatmap(
    corr_matrix,
    annot=True,
    fmt=".2f",
    cmap="RdBu_r",
    vmin=-1,
    vmax=1,
    center=0,
    square=True,
    linewidths=0.5,
    cbar_kws={"label": "Pearson correlation"}
)
plt.title("연속형 경제자원 변수와 개인정보 제공 동의 간 Pearson 상관관계")
plt.tight_layout()
plt.show()

# 범주형 소득구간(INCOME_CAT)은 각 소득구간별 개인정보 제공 동의 평균을 확인
income_cat_summary = (
    df.groupby("INCOME_CAT", observed=True)[["GIVEPIFR_R", "GIVEPIPC_R"]]
      .mean()
      .reset_index()
)
display(income_cat_summary)

plt.figure(figsize=(10, 5))
sns.lineplot(data=income_cat_summary, x="INCOME_CAT", y="GIVEPIFR_R", marker="o", label="GIVEPIFR_R")
sns.lineplot(data=income_cat_summary, x="INCOME_CAT", y="GIVEPIPC_R", marker="o", label="GIVEPIPC_R")
plt.title("INCOME_CAT 소득구간별 개인정보 제공 동의 평균")
plt.xlabel("INCOME_CAT 소득구간")
plt.ylabel("평균 동의 점수")
plt.tight_layout()
plt.show()

## 회귀분석 기본 가정 점검: 등분산성

회귀분석의 기본 가정 중 하나는 오차항의 분산이 모든 관측치에서 일정하다는 **등분산성**이다. 아래에서는 주 분석 모형과 동일한 변수 조합으로 예비 OLS를 적합한 뒤 Breusch-Pagan 검정과 White 검정을 실시한다.

해석 기준:
- p-value < 0.05: 등분산성 가정 기각, 이분산성 가능성 있음
- p-value >= 0.05: 등분산성 가정을 기각할 충분한 증거 없음

In [21]:
from statsmodels.stats.diagnostic import het_breuschpagan, het_white

# 주 분석 모형과 동일한 변수 조합으로 예비 OLS를 적합해 등분산성 검정
assumption_formula_main = (
    "{y} ~ log_INCOM0 + RANK + SATFIN_R + FINPROS_R + AGE "
    "+ C(SEX) + EDUC + ABLITINF_R + TECHHARM_R"
)

def heteroskedasticity_tests(formula, data, model_name):
    preliminary_ols = smf.ols(formula=formula, data=data).fit()

    bp_lm, bp_lm_p, bp_f, bp_f_p = het_breuschpagan(
        preliminary_ols.resid,
        preliminary_ols.model.exog
    )
    white_lm, white_lm_p, white_f, white_f_p = het_white(
        preliminary_ols.resid,
        preliminary_ols.model.exog
    )

    return {
        "model": model_name,
        "nobs": int(preliminary_ols.nobs),
        "bp_lm_stat": bp_lm,
        "bp_lm_pvalue": bp_lm_p,
        "bp_f_stat": bp_f,
        "bp_f_pvalue": bp_f_p,
        "white_lm_stat": white_lm,
        "white_lm_pvalue": white_lm_p,
        "white_f_stat": white_f,
        "white_f_pvalue": white_f_p,
        "bp_heteroskedasticity_5pct": bp_lm_p < 0.05,
        "white_heteroskedasticity_5pct": white_lm_p < 0.05,
    }

heteroskedasticity_rows = []
for y in dv_vars:
    heteroskedasticity_rows.append(
        heteroskedasticity_tests(
            assumption_formula_main.format(y=y),
            df,
            f"ASSUMPTION_OLS_{y}_logINCOM0_main"
        )
    )

heteroskedasticity_results = pd.DataFrame(heteroskedasticity_rows)
heteroskedasticity_results

,model,nobs,bp_lm_stat,bp_lm_pvalue,bp_f_stat,bp_f_pvalue,white_lm_stat,white_lm_pvalue,white_f_stat,white_f_pvalue,bp_heteroskedasticity_5pct,white_heteroskedasticity_5pct
0,ASSUMPTION_OLS_GIVEPIFR_R_logINCOM0_main,1067,14.831601,0.095667,1.655523,0.095295,69.171582,0.067092,1.324968,0.062804,False,False
1,ASSUMPTION_OLS_GIVEPIPC_R_logINCOM0_main,1067,11.197041,0.262444,1.245526,0.263071,52.973966,0.475168,0.998497,0.479126,False,False


## 주 모형: OLS/WLS

주 분석에서는 객관적 계층 변수를 `log_INCOM0`로 사용한다. 종속변수는 역코딩된 5점 척도(`GIVEPIFR_R`, `GIVEPIPC_R`)이며, 값이 높을수록 개인정보 제공 동의가 높다.

표본가중치 `FINALWT`가 있으므로 WLS를 기본으로 제시한다. 표준오차는 statsmodels의 기본 WLS/Logit 표준오차를 사용한다.

`log_INCOM0`는 월평균 가구소득 원자료(`INCOM0`)를 로그변환한 연속형 변수다. `INCOME`은 범주형 소득 변수이므로 여기서는 제외하고, 뒤의 강건성 분석에서 별도로 사용한다. `SEX`는 범주형으로, `EDUC`는 순서형 숫자 변수로 처리한다.

In [22]:
def fit_wls(formula, data, weight_col="FINALWT"):
    model_data = data.dropna(subset=[weight_col]).copy()
    res = smf.wls(formula=formula, data=model_data, weights=model_data[weight_col]).fit()
    return res

def fit_logit(formula, data):
    return smf.logit(formula=formula, data=data).fit(disp=False, maxiter=200)

def tidy_result(res, model_name):
    out = pd.DataFrame({
        "term": res.params.index,
        "coef": res.params.values,
        "std_err": res.bse.values,
        "p_value": res.pvalues.values,
        "ci_low": res.conf_int().iloc[:, 0].values,
        "ci_high": res.conf_int().iloc[:, 1].values,
    })
    out.insert(0, "model", model_name)
    return out

main_formula = (
    "{y} ~ log_INCOM0 + RANK + SATFIN_R + FINPROS_R + AGE "
    "+ C(SEX) + EDUC + ABLITINF_R + TECHHARM_R"
)

models = {}
for y in dv_vars:
    models[f"MAIN_WLS_{y}_logINCOM0"] = fit_wls(main_formula.format(y=y), df)

key_terms = ["log_INCOM0", "RANK", "SATFIN_R", "FINPROS_R", "ABLITINF_R", "TECHHARM_R"]

def is_key_term(term, extra_terms=None):
    extra_terms = extra_terms or []
    return term in key_terms or term in extra_terms

tidy_ols = pd.concat([tidy_result(res, name) for name, res in models.items()], ignore_index=True)
tidy_ols[tidy_ols["term"].apply(is_key_term)].sort_values(["model", "term"])

,model,term,coef,std_err,p_value,ci_low,ci_high
8,MAIN_WLS_GIVEPIFR_R_logINCOM0,ABLITINF_R,0.050658,0.037257,1.742164e-01,-0.022448,0.123763
5,MAIN_WLS_GIVEPIFR_R_logINCOM0,FINPROS_R,-0.044312,0.047656,3.526660e-01,-0.137823,0.049198
3,MAIN_WLS_GIVEPIFR_R_logINCOM0,RANK,0.060582,0.027121,2.570497e-02,0.007366,0.113799
4,MAIN_WLS_GIVEPIFR_R_logINCOM0,SATFIN_R,-0.110572,0.043439,1.105507e-02,-0.195808,-0.025335
9,MAIN_WLS_GIVEPIFR_R_logINCOM0,TECHHARM_R,0.049691,0.034441,1.493763e-01,-0.017889,0.117270
2,MAIN_WLS_GIVEPIFR_R_logINCOM0,log_INCOM0,-0.030171,0.054249,5.782152e-01,-0.136619,0.076276
18,MAIN_WLS_GIVEPIPC_R_logINCOM0,ABLITINF_R,0.030027,0.032937,3.621608e-01,-0.034602,0.094657
15,MAIN_WLS_GIVEPIPC_R_logINCOM0,FINPROS_R,-0.086803,0.042131,3.961178e-02,-0.169471,-0.004134
13,MAIN_WLS_GIVEPIPC_R_logINCOM0,RANK,0.056532,0.023976,1.856500e-02,0.009485,0.103578
14,MAIN_WLS_GIVEPIPC_R_logINCOM0,SATFIN_R,-0.091652,0.038403,1.717966e-02,-0.167007,-0.016298


In [23]:
# 전체 회귀표
for name, res in models.items():
    print("=" * 100)
    print(name)
    print(res.summary())


MAIN_WLS_GIVEPIFR_R_logINCOM0
                            WLS Regression Results                            
Dep. Variable:             GIVEPIFR_R   R-squared:                       0.059
Model:                            WLS   Adj. R-squared:                  0.051
Method:                 Least Squares   F-statistic:                     7.331
Date:                Mon, 18 May 2026   Prob (F-statistic):           2.05e-10
Time:                        10:15:59   Log-Likelihood:                -1633.8
No. Observations:                1067   AIC:                             3288.
Df Residuals:                    1057   BIC:                             3337.
Df Model:                           9                                         
Covariance Type:            nonrobust                                         
                    coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------
Intercept       

`GIVEPIFR_R`는 “할인이나 무료상품을 받기 위해 개인정보를 제공할 의향”이다. 역코딩했기 때문에 값이 높을수록 제공 동의가 높다.

주요 결과 해석:

- `log_INCOM0`: 계수 -0.030, p=0.578  
  객관적 가구소득은 할인/무료상품을 위한 개인정보 제공 동의와 통계적으로 유의한 관련이 없다.

- `RANK`: 계수 0.061, p=0.026  
  주관적 계층의식이 높을수록 제공 동의가 높다. 이는 “낮은 계층일수록 제공 동의가 높을 것”이라는 가설과는 반대 방향이다.

- `SATFIN_R`: 계수 -0.111, p=0.011  
  `SATFIN_R`은 높을수록 가계상태 만족도가 높은 변수다. 계수가 음수이므로, 가계상태 만족도가 높을수록 제공 동의는 낮다. 반대로 말하면 가계상태 만족도가 낮을수록 할인/무료상품을 위한 개인정보 제공 동의가 높다는 해석이 가능하다.

- `AGE`: 계수 -0.015, p<0.001  
  나이가 많을수록 제공 동의가 낮다.

- `FINPROS_R`, `EDUC`, `ABLITINF_R`, `TECHHARM_R`는 이 모형에서 통계적으로 유의하지 않다.

In [ ]:
# 통계적으로 유의미한 관계만 표시: p < 0.05
significant_ols = tidy_ols[tidy_ols["p_value"] < 0.05].copy()

# 절편은 제외
significant_ols = significant_ols[significant_ols["term"] != "Intercept"]

significant_ols = significant_ols.sort_values(["model", "p_value"])
significant_ols[["model", "term", "coef", "std_err", "p_value", "ci_low", "ci_high"]]

### 해석할 때 주의!

종속변수는 역코딩된 개인정보 제공 동의 점수이다. 즉 `GIVEPIFR_R`와 `GIVEPIPC_R`는 값이 높을수록 개인정보 제공에 더 동의한다.

따라서 선형회귀/WLS에서 계수가 양수이면 해당 독립변수가 증가할수록 개인정보 제공 동의가 높아지는 방향이고, 계수가 음수이면 개인정보 제공 동의가 낮아지는 방향이다.

소득, 나이, 성별, 학력, 정보능력(ABLITINF_R), 미래 전망(FINPROS_R) 등 여러 변수가 한꺼번에 포함되어 있으므로, 각 계수는 나머지 변수를 통제한 조건부 관계로 해석해야 한다.

## 표준화 계수 모형

핵심 독립변수의 상대적 크기를 비교하기 위해 연속형 변수들을 z-score로 표준화한 모형을 추정한다. 종속변수는 원래 1-5 척도를 유지한다.


In [ ]:
formula_z = (
    "{y} ~ z_log_INCOM0 + z_RANK + z_SATFIN_R + z_FINPROS_R + z_AGE "
    "+ C(SEX) + EDUC + z_ABLITINF_R + z_TECHHARM_R"
)

z_models = {f"Z_WLS_{y}": fit_wls(formula_z.format(y=y), df) for y in dv_vars}
tidy_z = pd.concat([tidy_result(res, name) for name, res in z_models.items()], ignore_index=True)
z_key_terms = ["z_log_INCOM0", "z_RANK", "z_SATFIN_R", "z_FINPROS_R", "z_AGE", "z_ABLITINF_R", "z_TECHHARM_R"]
tidy_z[tidy_z["term"].isin(z_key_terms)].sort_values(["model", "term"])


## 로짓모형: 개인정보 제공 의향 있음 여부

원척도 기준 `1=매우 동의`, `2=동의` 응답이면 1, `3`~`5`이면 0으로 이진화해 로짓모형을 추정한다. 계수는 개인정보 제공에 동의할 로그오즈의 변화로 해석하고, 평균 한계효과도 함께 계산한다.

In [24]:
logit_formula = (
    "{y}_agree ~ log_INCOM0 + RANK + SATFIN_R + FINPROS_R + AGE "
    "+ C(SEX) + EDUC + ABLITINF_R + TECHHARM_R"
)

logit_models = {}
for y in raw_dv_vars:
    logit_models[f"MAIN_LOGIT_{y}_agree_logINCOM0"] = fit_logit(logit_formula.format(y=y), df)

tidy_logit = pd.concat([tidy_result(res, name) for name, res in logit_models.items()], ignore_index=True)
tidy_logit[tidy_logit["term"].apply(is_key_term)].sort_values(["model", "term"])

,model,term,coef,std_err,p_value,ci_low,ci_high
8,MAIN_LOGIT_GIVEPIFR_agree_logINCOM0,ABLITINF_R,0.180538,0.087795,0.039748,0.008463,0.352613
5,MAIN_LOGIT_GIVEPIFR_agree_logINCOM0,FINPROS_R,-0.066540,0.109958,0.545085,-0.282054,0.148974
3,MAIN_LOGIT_GIVEPIFR_agree_logINCOM0,RANK,0.116302,0.061865,0.060116,-0.004951,0.237555
4,MAIN_LOGIT_GIVEPIFR_agree_logINCOM0,SATFIN_R,-0.178840,0.100859,0.076201,-0.376521,0.018840
9,MAIN_LOGIT_GIVEPIFR_agree_logINCOM0,TECHHARM_R,0.028877,0.080730,0.720573,-0.129351,0.187104
2,MAIN_LOGIT_GIVEPIFR_agree_logINCOM0,log_INCOM0,-0.024219,0.115869,0.834433,-0.251317,0.202880
18,MAIN_LOGIT_GIVEPIPC_agree_logINCOM0,ABLITINF_R,0.248278,0.118437,0.036056,0.016147,0.480410
15,MAIN_LOGIT_GIVEPIPC_agree_logINCOM0,FINPROS_R,-0.091139,0.149746,0.542773,-0.384637,0.202358
13,MAIN_LOGIT_GIVEPIPC_agree_logINCOM0,RANK,0.077151,0.080368,0.337065,-0.080366,0.234669
14,MAIN_LOGIT_GIVEPIPC_agree_logINCOM0,SATFIN_R,-0.144616,0.138421,0.296138,-0.415917,0.126685


In [25]:
# 평균 한계효과(average marginal effects)
## 독립변수가 1단위 증가할 때, 종속변수가 1이 될 확률이 평균적으로 얼마나 변하는가?로 해석할 수 있다. 
for name, res in logit_models.items():
    print("=" * 100)
    print(name)
    margeff = res.get_margeff(at="overall")
    print(margeff.summary())
    

MAIN_LOGIT_GIVEPIFR_agree_logINCOM0
        Logit Marginal Effects       
Dep. Variable:         GIVEPIFR_agree
Method:                          dydx
At:                           overall
                   dy/dx    std err          z      P>|z|      [0.025      0.975]
---------------------------------------------------------------------------------
C(SEX)[T.2.0]     0.0480      0.026      1.836      0.066      -0.003       0.099
log_INCOM0       -0.0040      0.019     -0.209      0.834      -0.041       0.033
RANK              0.0191      0.010      1.888      0.059      -0.001       0.039
SATFIN_R         -0.0294      0.017     -1.779      0.075      -0.062       0.003
FINPROS_R        -0.0109      0.018     -0.605      0.545      -0.046       0.024
AGE              -0.0039      0.001     -3.393      0.001      -0.006      -0.002
EDUC             -0.0241      0.012     -2.068      0.039      -0.047      -0.001
ABLITINF_R        0.0297      0.014      2.066      0.039       0.002     

In [26]:
# 로짓모형에서 통계적으로 유의미한 관계만 표시: p < 0.05
significant_logit = tidy_logit[tidy_logit["p_value"] < 0.05].copy()

# 절편은 연구질문 해석 대상이 아니므로 제외
significant_logit = significant_logit[significant_logit["term"] != "Intercept"]

significant_logit = significant_logit.sort_values(["model", "p_value"])
significant_logit[["model", "term", "coef", "std_err", "p_value", "ci_low", "ci_high"]]

,model,term,coef,std_err,p_value,ci_low,ci_high
6,MAIN_LOGIT_GIVEPIFR_agree_logINCOM0,AGE,-0.023715,0.007079,0.000807,-0.037589,-0.009841
7,MAIN_LOGIT_GIVEPIFR_agree_logINCOM0,EDUC,-0.146621,0.071247,0.039597,-0.286263,-0.006980
8,MAIN_LOGIT_GIVEPIFR_agree_logINCOM0,ABLITINF_R,0.180538,0.087795,0.039748,0.008463,0.352613
19,MAIN_LOGIT_GIVEPIPC_agree_logINCOM0,TECHHARM_R,0.430535,0.107612,0.000063,0.219620,0.641450
18,MAIN_LOGIT_GIVEPIPC_agree_logINCOM0,ABLITINF_R,0.248278,0.118437,0.036056,0.016147,0.480410


## 결과 저장

핵심 회귀계수 표를 CSV로 저장한다.

> 필요하면 주석 풀고 다운 받으세요!!


In [ ]:
# OUT_DIR = DATA_PATH.parent

# result_tables = {
#     "privacy_heteroskedasticity_tests.csv": heteroskedasticity_results,
#     "privacy_wls_core_coefficients.csv": tidy_ols,
#     "privacy_wls_standardized_coefficients.csv": tidy_z,
#     "privacy_logit_coefficients.csv": tidy_logit,
# }

# for filename, table in result_tables.items():
#     path = OUT_DIR / filename
#     table.to_csv(path, index=False, encoding="utf-8-sig")
#     print(path)


## 해석 가이드

- 주 분석에서 연구질문에 대한 핵심 검정은 `log_INCOM0` 계수다.
- 선형회귀/WLS의 종속변수는 역코딩된 개인정보 제공 동의 점수다. 값이 높을수록 개인정보 제공 동의가 높다.
- `log_INCOM0` 계수가 음수이고 통계적으로 유의하면, 월평균 가구소득이 낮을수록 개인정보 제공 동의가 높다는 가설과 부합한다.
- `RANK`는 1=최하층, 10=최상층이므로 계수가 음수이면 주관적 계층의식이 낮을수록 개인정보 제공 동의가 높다는 방향이다.
- `SATFIN_R`은 1=매우 불만족, 5=매우 만족이므로 계수가 음수이면 가계상태 만족도가 낮을수록 개인정보 제공 동의가 높다는 방향이다.
- `FINPROS_R`은 1=나빠질 것, 5=좋아질 것이므로 계수가 음수이면 가계경제 전망이 부정적일수록 개인정보 제공 동의가 높다는 방향이다.
- `ABLITINF_R`은 1=매우 낮음, 5=매우 높음이므로 계수가 양수이면 정보판별능력이 높을수록 개인정보 제공 동의가 높다는 방향이다.
- 범주형 소득 변수 `INCOME`은 주 분석의 대체 변수가 아니라, 마지막 강건성 분석에서 소득 측정 방식을 바꿔도 결과가 유지되는지 확인하는 용도로 해석한다.

## 축약 통제변수 모형: AGE, SEX, EDUC만 포함

아래 모형은 통제변수를 `control_vars = ["AGE", "SEX", "EDUC"]`로 제한한 버전이다. 주 분석과 동일하게 객관적 계층 변수는 `log_INCOM0`를 사용하고, 핵심 독립변수(`log_INCOM0`, `RANK`, `SATFIN_R`, `FINPROS_R`)는 유지한다.

회귀식은 다음과 같다.

$$
Y_i = \beta_0 + \beta_1 \log(\mathrm{INCOM0}_i + 1) + \beta_2 \mathrm{RANK}_i + \beta_3 \mathrm{SATFIN\_R}_i + \beta_4 \mathrm{FINPROS\_R}_i + \delta_1 \mathrm{AGE}_i + \delta_2 \mathrm{SEX}_i + \delta_3 \mathrm{EDUC}_i + \epsilon_i
$$

In [ ]:
# 축약 통제변수: AGE, SEX, EDUC만 사용
control_vars_basic = ["AGE", "SEX", "EDUC"]

formula_basic_main = (
    "{y} ~ log_INCOM0 + RANK + SATFIN_R + FINPROS_R + AGE "
    "+ C(SEX) + EDUC"
)

basic_models = {}
for y in dv_vars:
    basic_models[f"BASIC_WLS_{y}_logINCOM0"] = fit_wls(formula_basic_main.format(y=y), df)

tidy_basic = pd.concat([tidy_result(res, name) for name, res in basic_models.items()], ignore_index=True)
basic_key_terms = ["log_INCOM0", "RANK", "SATFIN_R", "FINPROS_R", "AGE", "C(SEX)[T.2.0]", "EDUC"]
tidy_basic[tidy_basic["term"].apply(lambda x: is_key_term(x, basic_key_terms))].sort_values(["model", "term"])

In [ ]:
# 축약 통제변수 로짓모형: 개인정보 제공 의향 있음(원척도 1 또는 2) 여부
logit_formula_basic = (
    "{y}_agree ~ log_INCOM0 + RANK + SATFIN_R + FINPROS_R + AGE "
    "+ C(SEX) + EDUC"
)

basic_logit_models = {}
for y in raw_dv_vars:
    basic_logit_models[f"BASIC_LOGIT_{y}_agree_logINCOM0"] = fit_logit(logit_formula_basic.format(y=y), df)

tidy_basic_logit = pd.concat([tidy_result(res, name) for name, res in basic_logit_models.items()], ignore_index=True)
tidy_basic_logit[tidy_basic_logit["term"].apply(lambda x: is_key_term(x, basic_key_terms))].sort_values(["model", "term"])

In [ ]:
# 축약 통제변수 로짓모형의 평균 한계효과
for name, res in basic_logit_models.items():
    print("=" * 100)
    print(name)
    margeff = res.get_margeff(at="overall") 
    print(margeff.summary())


In [ ]:
# 축약 통제변수 로짓모형에서 통계적으로 유의미한 관계만 표시: p < 0.05
significant_basic_logit = tidy_basic_logit[tidy_basic_logit["p_value"] < 0.05].copy()

# 절편은 연구질문 해석 대상이 아니므로 제외
significant_basic_logit = significant_basic_logit[significant_basic_logit["term"] != "Intercept"]

significant_basic_logit = significant_basic_logit.sort_values(["model", "p_value"])
significant_basic_logit[["model", "term", "coef", "std_err", "p_value", "ci_low", "ci_high"]]


In [ ]:
# # 축약 통제변수 모형 결과 저장
# basic_wls_path = OUT_DIR / "privacy_basic_controls_wls_coefficients.csv"
# basic_logit_path = OUT_DIR / "privacy_basic_controls_logit_coefficients.csv"

# tidy_basic.to_csv(basic_wls_path, index=False, encoding="utf-8-sig")
# tidy_basic_logit.to_csv(basic_logit_path, index=False, encoding="utf-8-sig")

# print(basic_wls_path)
# print(basic_logit_path)


## 한국어 회귀식

축약 통제변수 모형의 회귀식은 한국어로 다음과 같이 표현할 수 있다. 여기서 종속변수는 역코딩된 개인정보 제공 동의 점수이며, 값이 높을수록 제공 동의가 높다.

$$
\text{개인정보제공동의}_i
=
\beta_0
+
\beta_1 \text{로그월평균가구소득}_i
+
\beta_2 \text{주관적계층의식}_i
+
\beta_3 \text{가계상태만족도\_역코딩}_i
+
\beta_4 \text{가계경제전망\_역코딩}_i
+
\delta_1 \text{연령}_i
+
\delta_2 \text{성별}_i
+
\delta_3 \text{교육수준}_i
+
\epsilon_i
$$

문장으로 풀어 쓰면, 개인 $i$의 개인정보 제공 동의는 로그 월평균 가구소득, 주관적 계층의식, 역코딩된 가계상태 만족도(`SATFIN_R`), 역코딩된 가계경제 전망(`FINPROS_R`)의 영향을 받으며, 연령, 성별, 교육수준을 통제한 회귀모형으로 추정한다.

여기서 `개인정보제공동의`는 분석에 따라 `GIVEPIFR_R` 또는 `GIVEPIPC_R`를 의미한다.

## 독립변수 다중공선성 점검: VIF

다중공선성은 종속변수와 독립변수의 관계가 아니라 **독립변수들끼리 강하게 관련되어 있는지**의 문제다. 따라서 실제 주 분석 회귀식에 동시에 들어가는 독립변수들을 대상으로 VIF(Variance Inflation Factor)를 계산한다.

해석 기준은 일반적으로 다음과 같다.

- VIF 1: 공선성 거의 없음
- VIF 2~5: 어느 정도 관련은 있으나 대체로 허용 가능
- VIF 5 이상: 주의 필요
- VIF 10 이상: 심각한 다중공선성 가능성

여기서는 주 분석 모형에 맞춰 `log_INCOM0`, `RANK`, `SATFIN_R`, `FINPROS_R`, 통제변수의 VIF를 확인한다. `INCOME_CAT`은 강건성 분석에서 별도로 사용한다.

In [ ]:
from patsy import dmatrices
from statsmodels.stats.outliers_influence import variance_inflation_factor

# 실제 주 회귀식과 같은 독립변수 조합으로 VIF 계산
vif_formula_main = (
    "GIVEPIFR_R ~ log_INCOM0 + RANK + SATFIN_R + FINPROS_R + AGE "
    "+ C(SEX) + EDUC + ABLITINF_R + TECHHARM_R"
)

def calculate_vif(formula, data, model_name):
    # dmatrices를 쓰면 C(SEX) 같은 formula 항을 회귀식과 동일하게 설계행렬로 변환할 수 있다.
    y, X = dmatrices(formula, data=data, return_type="dataframe")

    rows = []
    for i, col in enumerate(X.columns):
        if col == "Intercept":
            continue
        rows.append({
            "model": model_name,
            "term": col,
            "vif": variance_inflation_factor(X.values, i),
        })
    out = pd.DataFrame(rows).sort_values("vif", ascending=False).reset_index(drop=True)
    return out

vif_results = calculate_vif(vif_formula_main, df, "VIF_main_logINCOM0_model")
vif_results
# 해보니까 문제 없네요~

In [ ]:
# # VIF 결과 저장
# vif_path = OUT_DIR / "privacy_vif_multicollinearity.csv"
# vif_results.to_csv(vif_path, index=False, encoding="utf-8-sig")
# print(vif_path)


## 강건성 분석: 범주형 소득 변수 INCOME

주 분석의 객관적 계층 변수는 `log_INCOM0`이다. 아래에서는 소득 측정 방식을 바꿔도 결과가 유지되는지 확인하기 위해, 월평균 가구소득 범주형 변수 `INCOME`을 인접 구간으로 묶은 `INCOME_CAT`을 사용한다.

`INCOME_CAT`은 범주형 변수이므로 `C(INCOME_CAT)`으로 투입하며, 계수는 기준 소득구간 대비 각 소득구간의 차이로 해석한다. 이 분석은 주 분석을 대체하는 것이 아니라 강건성 검토다.


In [ ]:
robust_formula_income_cat = (
    "{y} ~ C(INCOME_CAT) + RANK + SATFIN_R + FINPROS_R + AGE "
    "+ C(SEX) + EDUC + ABLITINF_R + TECHHARM_R"
)

robust_models = {}
for y in dv_vars:
    robust_models[f"ROBUST_WLS_{y}_INCOME_cat"] = fit_wls(robust_formula_income_cat.format(y=y), df)

robust_tidy_ols = pd.concat([tidy_result(res, name) for name, res in robust_models.items()], ignore_index=True)
robust_tidy_ols[
    robust_tidy_ols["term"].apply(lambda x: x.startswith("C(INCOME_CAT)") or x in key_terms)
].sort_values(["model", "term"])


In [ ]:
robust_logit_formula = (
    "{y}_agree ~ C(INCOME_CAT) + RANK + SATFIN_R + FINPROS_R + AGE "
    "+ C(SEX) + EDUC + ABLITINF_R + TECHHARM_R"
)

robust_logit_models = {}
for y in raw_dv_vars:
    robust_logit_models[f"ROBUST_LOGIT_{y}_agree_INCOME_cat"] = fit_logit(robust_logit_formula.format(y=y), df)

robust_tidy_logit = pd.concat([tidy_result(res, name) for name, res in robust_logit_models.items()], ignore_index=True)
robust_tidy_logit[
    robust_tidy_logit["term"].apply(lambda x: x.startswith("C(INCOME_CAT)") or x in key_terms)
].sort_values(["model", "term"])

In [ ]:
# 강건성 로짓모형의 평균 한계효과
for name, res in robust_logit_models.items():
    print("=" * 100)
    print(name)
    margeff = res.get_margeff(at="overall")
    print(margeff.summary())


In [ ]:
# 강건성 분석 모형의 VIF
robust_vif_formula = (
    "GIVEPIFR_R ~ C(INCOME_CAT) + RANK + SATFIN_R + FINPROS_R + AGE "
    "+ C(SEX) + EDUC + ABLITINF_R + TECHHARM_R"
)

robust_vif_results = calculate_vif(robust_vif_formula, df, "VIF_robust_INCOME_cat_model")
robust_vif_results

## 교수님께 질문하고 싶은 부분 & 논의가 필요한 부분

**1. 회귀 모형(+ 가중회귀 모형)과 로짓 모형 중에 어떤 방법론을 선택하는게 타당할까(각 방법의 이유가 있을거 같은데...)** 

2. 상관관계는 매우 낮은데, 회귀 계수가 유의미하게 나온 상황에서 어떻게 해석해야 하는가? -> ai 한테 물어보거나, ai한테 이런 실제 연구 줘! -> 하다가 모르겠으면 질문... 

 
3. income 과 logincom0 변수를 어떻게 처리할지.. 따로따로? 아니면 둘 다?  -> **교수님**
> INCOM0: 월평균 가구소득(원료/연속형)
> INCOME: 월평균가구소득(범주형)

즉, 가구소득으로 했음. 개인소득으로 해야할까..? -> 이게 제일 큰 한계일듯 








---


4. 종속변수 두 개? -> 내가 봤을 때는 GIVEPIFR: 개인정보제공에대한태도1:할인이나무료상품을받기위해제공할것  -> 주로 이거를 쓰고 후속 분석을 위해 GIVEPIPC는 남겨두자 
GIVEPIPC: 개인정보제공에대한태도2:민간기업이나의개인정보를사용하여이익을내더라도제공할것  

중에서 GIVEPIFR는 무조건인데...
> 왜냐하면 우리가 보고 싶은 것은 혜택의 의미이기 떄문에 민간기업이 나의 개인정보로 이익을 내더라도 줄 것은 나에게 곧바로 혜택이 돌아오는 차원의 이야기는 아니니까. 종속변수 하나로 가야하맂 않을까?

WLS로 하자. / 회귀분석으로 하자

상관관계는 매우 낮은데 회귀 계수가 유의미하게 나온 상황에서 어떻게 해석해야 하는가? -> 사회과학에서 아주 낮은 R**2가 나오더라더, 회귀계수 위주로 해석한다. 